test/forge/Strategies.sol

```diff
@@ -0,0 +1,1829 @@
+        uint256 strategyId = carbonController.createStrategy{ value: val }(tokens[i0], tokens[i1], [order, order]);
+
+        // get balances before
+        // 0 -> t0 user1
+        // 1 -> t1 user1
+        // 2 -> t0 controller
+        // 3 -> t1 controller
+        uint256[4] memory balancesBefore = [
+            tokens[i0].balanceOf(user1),
+            tokens[i1].balanceOf(user1),
+            tokens[i0].balanceOf(address(carbonController)),
+            tokens[i1].balanceOf(address(carbonController))
+        ];
+
+        // create new orders
+        Order memory order0 = updateOrderDelta(order, deltas[0]);
+        Order memory order1 = updateOrderDelta(order, deltas[1]);
+
+        // update strategy
+        val = tokens[i0] == NATIVE_TOKEN && deltas[0] >= 0 ? abs(deltas[0]) : 0;
+        val += tokens[i1] == NATIVE_TOKEN && deltas[1] >= 0 ? abs(deltas[1]) : 0;
+        // add excess native token
+        val += 1 ether;
+        carbonController.updateStrategy{ value: val }(strategyId, [order, order], [order0, order1]);
+
+        // get balances after
+        // 0 -> t0 user1
+        // 1 -> t1 user1
+        // 2 -> t0 controller
+        // 3 -> t1 controller
+        uint256[4] memory balancesAfter = [
+            tokens[i0].balanceOf(user1),
+            tokens[i1].balanceOf(user1),
+            tokens[i0].balanceOf(address(carbonController)),
+            tokens[i1].balanceOf(address(carbonController))
+        ];
+
+        // user1 balance should decrease by delta amount and controller should increase if delta is positive
+        // user1 balance should increase by delta amount and controller should decrease if delta is negative
+        if (deltas[0] >= 0) {
+            assertEq(balancesAfter[0], balancesBefore[0] - abs(deltas[0]));
+            assertEq(balancesAfter[2], balancesBefore[2] + abs(deltas[0]));
+        } else {
+            assertEq(balancesAfter[0], balancesBefore[0] + abs(deltas[0]));
+            assertEq(balancesAfter[2], balancesBefore[2] - abs(deltas[0]));
+        }
+        if (deltas[1] >= 0) {
+            assertEq(balancesAfter[1], balancesBefore[1] - abs(deltas[1]));
+            assertEq(balancesAfter[3], balancesBefore[3] + abs(deltas[1]));
+        } else {
+            assertEq(balancesAfter[1], balancesBefore[1] + abs(deltas[1]));
+            assertEq(balancesAfter[3], balancesBefore[3] - abs(deltas[1]));
+        }
+
+        vm.stopPrank();
+    }
+
+    /// @dev test that the strategy update reverts if the reference tokens are not equal to the current
+    function testStrategyUpdateRevertsIfTheProvidedReferenceTokensAreNotEqualToTheCurrent(
+        uint256 valueToChange
+    ) public {
+        vm.startPrank(user1);
+        valueToChange = bound(valueToChange, 0, 3);
+
+        Order memory order = generateTestOrder();
+        // create strategy
+        uint256 strategyId = carbonController.createStrategy(token0, token1, [order, order]);
+
+        if (valueToChange == 0) {
```

Comment:
I suggest passing an array of delta values as input.
You could then pass, for example `[0, 1, 0, 0]`, which would mimic what you're doing here.
But it will allow you to extend this test in the future, and pass an array of delta values which isn't necessarily made of one "1" and three "0".
In fact, you can even use an `Order` struct for these delta values, making it clear which delta value is designated for which field.

Reply:
Replaced the `valueToChange` with an Order struct and simply added the deltas - also increased the number of values for the deltas from 0 to 100 000.